File: src/main/java/dev/elysium/eraces/updaters/AdditionalArmorUpdater.java
package dev.elysium.eraces.updaters;

import dev.elysium.eraces.datatypes.Race;
import dev.elysium.eraces.updaters.base.IUpdater;
import org.bukkit.attribute.Attribute;
import org.bukkit.attribute.AttributeInstance;
import org.bukkit.entity.Player;

public class AdditionalArmorUpdater implements IUpdater {
    @Override
    public void update(Race race, Player player) {
        double additionalArmor = race.getAdditionalArmor();

        AttributeInstance attribute = player.getAttribute(Attribute.ARMOR);
        if (attribute != null)
            attribute.setBaseValue(additionalArmor);
    }
}

File: src/main/java/dev/elysium/eraces/updaters/AdditionalArmorUpdater.kt
package dev.elysium.eraces.updaters

import dev.elysium.eraces.datatypes.Race
import dev.elysium.eraces.updaters.base.IDisableable
import dev.elysium.eraces.updaters.base.IUpdater
import org.bukkit.attribute.Attribute
import org.bukkit.entity.Player

class AdditionalArmorUpdater : IUpdater, IDisableable {
    override fun update(race: Race, player: Player) {
        player.getAttribute(Attribute.ARMOR)?.baseValue = race.additionalArmor
    }

    override fun disable(player: Player) {
        player.getAttribute(Attribute.ARMOR)?.baseValue = 20.0
    }
}
